### DISCUSSION TITLE
20240208 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by cgbosse** [nautilus-cyberneering/.github#81] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - 
- MVanDijke(@WarmBeer) - 

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Tracker

Issues:

- [x] https://github.com/torrust/torrust-tracker/issues/639

Open issues:

- [ ] https://github.com/torrust/torrust-tracker/issues/669

Tracker Checker is fully operative with (only) Torrust Tracker:

```console
TORRUST_CHECKER_CONFIG='{
    "udp_trackers": ["144.126.245.19:6969"],
    "http_trackers": ["https://tracker.torrust-demo.com"],
    "health_checks": ["https://tracker.torrust-demo.com/api/health_check"]
}' cargo run --bin tracker_checker

Running checks for trackers ...
UDP trackers ...
✓ - Announce at 127.0.0.1:6969 is OK
✓ - Scrape at 127.0.0.1:6969 is OK
HTTP trackers ...
✓ - Announce at http://127.0.0.1:7070/ is OK
✓ - Scrape at http://127.0.0.1:7070/ is OK
Health checks ...
✓ - Health API at http://127.0.0.1:1313/health_check is OK
```

but we have to improve [a lot of things in the future](https://github.com/torrust/torrust-tracker/issues/669).

### Torrust Compose

- [ ] https://github.com/torrust/torrust-compose/issues/6

Fixed Nginx configuration for the image proxy. Updated on the live demo. [Final config](https://github.com/torrust/torrust-compose/blob/main/droplet/share/container/default/config/nginx.conf#L1-L36). I think the key point was you have to use `proxy_pass` without URL path (`http://index:3001`) and I guess you can "rewrite" the URL to remove the "api" but the extracted variable "$1" can't have any other variable. As soon as you manipulate the URL, it's decoded and normalized.

```
rewrite ^/api(/.*) $1 break;
proxy_pass http://index:3001;
```

Anyway, I think we should simplify the proxy to avoid this complexity. We could:

- Change the URL as we started to discuss.
- Use the "api" prefix for the api so that the api can be deployed in a subfolder. I think it's a pretty common way to do it.

### Torrust Index

Issues:

- [ ] https://github.com/torrust/torrust-index/issues/453
- [ ] https://github.com/torrust/torrust-index/issues/468
- [ ] https://github.com/torrust/torrust-index/issues/467

New console command to upload random torrents to the Index.

```console
cargo run --bin seeder -- --api-base-url "index.torrust-demo.com/api" --number-of-torrents 2419200 --user admin --password xxxxxxxxxxxx --interval 1
```

The mini server is having trouble handling UDP tracker requests. See https://github.com/torrust/torrust-index/issues/459#issuecomment-1932499213

Two problems detected:

- Seeder (command to upload torrents) needs a timeout.
- The statistics importer only imports 1 torrent statistic per second from the tracker. And it seems sometimes is stopped.

I need to:

- [Add more logs](https://github.com/torrust/torrust-index/issues/468).
- [Review problems with timeouts like in the Tracker](https://github.com/torrust/torrust-index/issues/467).
- [Improve the importer](https://github.com/torrust/torrust-index/issues/469)

Discussion:

- How to avoid DDoS attacks? API rate limit per user?
- [ ] https://github.com/torrust/torrust-index/issues/469

**Comment by mario-nt** [comment]
### GUI

- https://github.com/torrust/torrust-index-gui/issues/401
	

> -opened discussions


### Index

- https://github.com/torrust/torrust-index/issues/445
	

> [-new error message
> 	-refactor and fix PR tests not passing]([url](url))

	
- https://github.com/torrust/torrust-index/issues/446

> 	- research on optional extractors
> 	- worked on creating the new extractor




### Training/Research

  

>    - Optional extractors

**Comment by mickvandijke** [comment]
Did nothing this week.
